clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3151,11 +3151,14 @@ static void handleWorkGroupSize(Sema &S, Decl *D, const ParsedAttr &AL) {
       return;
     ZDimExpr = ZDim.get();
 
+    // If the declaration has an [[intel::num_simd_work_items()]] attribute,
+    // check to see if the last argument can be evenly divided by the
+    // num_simd_work_items attribute
     if (const auto *A = D->getAttr<SYCLIntelNumSimdWorkItemsAttr>()) {
       int64_t NumSimdWorkItems =
           A->getValue()->getIntegerConstantExpr(Ctx)->getSExtValue();
 
-      if (XDimVal.getZExtValue() % NumSimdWorkItems != 0) {
+      if (ZDimVal.getZExtValue() % NumSimdWorkItems != 0) {
```

Comment:
This check is called for `__attribute__((reqd_work_group_size))` as well as `[[intel::reqd_work_group_size]]` -- aren't X and Z only swapped for the intel attribute and not the OpenCL ones?

Reply:
I have added a check to verify SYCL vs OpenCL spelling